terraso_backend/apps/graphql/schema/users.py

```diff
@@ -107,3 +123,44 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
             )
 
         return super().mutate_and_get_payload(root, info, **kwargs)
+
+
+class UserPreferenceUpdate(relay.ClientIDMutation):
+    preference = graphene.Field(UserPreferenceNode)
+
+    model_class = UserPreference
+
+    class Input:
+        user_email = graphene.String(required=True)
+        key = graphene.String(required=True)
+        value = graphene.String(required=True)
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        request_user = info.context.user
+        user_email = kwargs.pop("user_email")
+        key = kwargs.pop("key")
+        value_raw = kwargs.pop("value")
+        user = User.objects.get(email=user_email)
+        preference, _ = UserPreference.objects.get_or_create(user_id=user.id, key=key)
+
+        if not rules.test_rule("allowed_to_update_preferences", request_user, preference):
+            logger.error(
+                "Attempt to update a User preferences, not allowed",
+                extra={"request_user_id": request_user.id, "target_user_id": user.id},
+            )
+            raise GraphQLNotAllowedException(
+                model_name=UserPreference.__name__, operation=MutationTypes.UPDATE
+            )
+
+        try:
+            value_json = json.loads(value_raw)
+        except json.JSONDecodeError:
+            raise GraphQLValidationException.from_custom_errors(
```

Comment:
Hey @cacarrara How does this look to you? I changed the `value` field to a json field and I had to add this validation, there are some decisions that I am not completely sure are correct:
- The `bad_format`code is hardcoded here, should that exist in a constant file? or where are the other codes?
- I create a `from_custom_errors` function to handle this error, is that the correct way? or is there something else in the code to handle something like this?